### DISCUSSION TITLE
Mesh Adaptivity for a set of vector variables

### DISCUSSION THREAD

**Discussion by lkuna24** [idaholab/moose#28292] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi,
I am working on simulating polarization in ferroelectric thin films utilizing and was wondering if it is possible to have a marker operate on a set of vector variables (polar_x, polar_y, polar_z) or possibly even the norm of the vector represented as an AuxVariable (something along the lines of VectorThresholdMarker). 

 I have been unsuccessful trying to create such a custom Marker from scratch as when I try to add the three polar variables, as *coupledValue*, the compile throws an error that it does not recognize *coupledValue*.  

In these simulations I need a fine mesh at the start for polar domains to form from the paralectric state and as they materialize, I would like for the mesh to coarsen within each domain, shown below:
![image](https://github.com/user-attachments/assets/f6b07a54-7fd3-4095-b68d-5961401739b1)

In this case pictured the coarsening is happening on the polar_z variable.  I can't just use a ComboMarker as all three polar variables exist on every point in the mesh and if the marker operating on polar_z marks for COARSEN, the markers for polar_x and polar_y will mark for REFINE and my understanding is that REFINE will always take precedence over COARSEN.  

Additionally I was wondering if max_h_level can be block restricted.

Thanks

**Comment by maxnezdyur** [comment]
I would use https://mooseframework.inl.gov/source/auxkernels/VectorVariableComponentAux.html or https://mooseframework.inl.gov/source/auxkernels/VectorVariableMagnitudeAux.html , which will get you an auxvariable that other markers would work on. Or use `coupledVectorValue` to get the vector variable in your custom marker.

**Reply by lkuna24** [reply]
The formulation used in the Ferret application utilizes three separate LAGRANGE variables for each vector component as opposed to one variable that is considered a vector.  I don't believe this approach would work for use in these simulations, but what Marker can utilize an AuxVariable instead of a Variable?

**Reply by maxnezdyur** [reply, marked as answer]
Think I am a bit lost.

> I am working on simulating polarization in ferroelectric thin films utilizing and was wondering if it is possible to have a marker operate on a set of vector variables (polar_x, polar_y, polar_z) or possibly even the norm of the vector represented as an AuxVariable (something along the lines of VectorThresholdMarker).

So it's not a "Vector Variable", instead you have 3 "regular" variables?  There are a few different auxkernels that can perform operations on sets of variables. This will give you a single variable that you can use for the markers. Markers work on auxvariables and variables.

**Reply by lkuna24** [reply]
Theres actually an AuxKernel inside of Ferret that computes the magnitude of the polarization using the three variable inputs.  Seems like this is working fine with the ValueThreshold Marker.  
Thanks for your suggestions!
